sem/runner.py

```diff
@@ -195,12 +195,21 @@ def get_available_parameters(self):
                              '(.*)General\sArguments.*',
                              result, re.DOTALL)
 
+        global_options = subprocess.check_output([self.script_executable,
+                                                  '--PrintGlobals'],
+                                                 env=self.environment,
+                                                 cwd=self.path).decode('utf-8')
+
         # Get the single parameter names
+        args = []
         if len(options):
-            args = re.findall('.*--(.*?):.*', options[0], re.MULTILINE)
-            return sorted(args)  # Return a sorted list
-        else:
-            return []
+            args += re.findall('.*--(.*?)[?::|=].*', options[0], re.MULTILINE)
+        if len(global_options):
+            args += [k for k in re.findall('.*--(.*?)[?::|=].*',
+                                           global_options, re.MULTILINE) if k
+                     not in ['RngRun', 'RngSeed', 'SchedulerType',
```

Comment:
I can try to automatically detect global variables, but I would still need to be guaranteed that the script I get them from will always only contain the default global variables, so that the defaults I infer are correct. If there is an alternative and more natural solution, I would gladly implement it!